src/mame/ensoniq/esq5505.cpp

```diff
@@ -294,15 +308,160 @@ class esq5505_state : public driver_device
 	void cpu_space_map(address_map &map) ATTR_COLD;
 	void eps_cpu_space_map(address_map &map) ATTR_COLD;
 
+	void update_floppy_inputs();
+	void floppy_loaded(bool loaded);
+	void floppy_load(floppy_image_device *floppy);
+	void floppy_unload(floppy_image_device *floppy);
+	void cartridge_loaded(bool loaded);
+	void cartridge_load(ensoniq_vfx_cartridge *cart);
+	void cartridge_unload(ensoniq_vfx_cartridge *cart);
+
+	void update_docirq_to_maincpu();
+	void otis_irq(int irq);
+
 	uint16_t m_analog_values[8];
 };
 
+void esq5505_state::cartridge_loaded(bool loaded)
+{
+	LOG("Cartridge %s\n", loaded ? "Inserted" : "Ejected");
+	int state = loaded ? CLEAR_LINE : ASSERT_LINE;
+
+	// On VFX and later, DUART input bit 1 is 0 for cartridge present.
+	LOG("ip1 -> %d\n", state);
+	m_duart->ip1_w(state);
+}
+
+void esq5505_state::cartridge_load(ensoniq_vfx_cartridge *cart)
+{
+	(void) cart;
```

Comment:
No need to explicitly void out unused function parameters, I believe any warnings that could be caused have been suppressed since forever.